app/src/main/java/com/example/event_finder_capstone/RetrofitClient.java

```diff
@@ -0,0 +1,47 @@
+package com.example.event_finder_capstone;
+
+import okhttp3.Interceptor;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import retrofit2.Retrofit;
+import retrofit2.converter.gson.GsonConverterFactory;
+
+public class RetrofitClient {
+
+    public static final String YELP_API_BASE_URL = "https://api.yelp.com/v3/";
+    public static final String value = "Bearer " + BuildConfig.YELP_KEY;
+    private static RetrofitClient instance = null;
+    private final EventEndpointsInterface Yelp_Api;
+
+    private RetrofitClient() {
+
+        Interceptor interceptor = chain -> {
+            Request newRequest = chain.request().newBuilder().addHeader("Authorization", value).build();
+            return chain.proceed(newRequest);
+        };
+
+        OkHttpClient.Builder builder = new OkHttpClient.Builder();
+        builder.interceptors().add(interceptor);
+        OkHttpClient client = builder.build();
+
+        Retrofit retrofit = new Retrofit.Builder()
+                .baseUrl(YELP_API_BASE_URL)
+                .addConverterFactory(GsonConverterFactory.create())
+                .client(client)
+                .build();
+
+        Yelp_Api = retrofit.create(EventEndpointsInterface.class);
+    }
+
+    public static synchronized RetrofitClient getInstance() {
```

Comment:
❓ Question:  Are you familiar with what this pattern is called and why it's used?

Reply:
I think I have a basic understanding that synchronized methods protect resource access to things accessed at the same time - would love to discuss this more during our 1:1!